content/en/docs/Reference/file-based-catalogs.md

```diff
@@ -0,0 +1,586 @@
+
+### Composability
+
+File-based catalogs are stored in an arbitrary directory hierarchy, which enables index composition. If I have two
+separate file-based catalog directories, `indexA` and `indexB`, I can make a new combined index by making a new
+directory `indexC` and copying `indexA` and `indexB` into it.
+
+This composability enables decentralized indexes. The format permits operator authors to maintain operator-specific
+indexes and index maintainers to trivially build an index composed of individual operator indexes.
+
+> NOTE: Duplicate packages and duplicate bundles within a package are not permitted. The `opm validate` command will
+> return an error if any duplicates are found.
+
+Since operator authors are most familiar with their operator, its dependencies, and its upgrade compatibility, they are
+able to maintain their own operator-specific index and have direct control over its contents. With file-based catalogs,
+operator authors own the task of building and maintaining their packages in an index. Composite index maintainers treat
+packages as a black box; they own the task of curating the packages in their catalog and publishing the catalog to
+users. 
+
+File-based catalogs can be composed by combining multiple other catalogs or by extracting subsets of one catalog, or a
+combination of both of these.
+
+### Extensibility
+
+The final design goal is to provide extensibility around indexes. The file-based catalog spec is a low-level
+representation of an index. While it can be maintained directly in its low-level form, we expect many index maintainers
+to build interesting extensions on top that can be used by their own custom tooling to make all sorts of mutations. For
+example, one could imagine a tool that translates a high-level API like (mode=semver) down to the low-level file-based
+catalog format for upgrade edges. Or perhaps an index maintainer needs to customize all of the bundle metadata by adding
+a new property to bundles that meet a certain criteria.
+
+The OLM developer community will be making use of this extensibility to build more official tooling on top of the
+low-level APIs, but the major benefit is that index maintainers have this capability as well.
+
+## Specification
+
+### Structure
+
+File-based catalogs can be stored and loaded from directory-based filesystems.
+
+`opm` loads the catalog by walking the root directory and recursing into subdirectories. It attempts to load every file
+it finds and fails if any errors occur.
+
+Non-catalog files can be ignored using `.indexignore` files, which behave identically to `.gitignore` files. That is,
+they have the same rules for [patterns](https://git-scm.com/docs/gitignore#_pattern_format) and precedence.
+
+> #### Example `.gitignore` file
+> ```gitignore
+> # Ignore everything except non-object .json and .yaml files
+> **/*
+> !*.json
+> !*.yaml
+> **/objects/*.json
+> **/objects/*.yaml
+> ```
```

Comment:
This seems to render weirdly as well on my local browser:

![Screenshot from 2021-09-02 13-46-47](https://user-images.githubusercontent.com/9899409/131892322-3b6a001b-96b2-49e5-b6aa-1631f3346237.png)